### PR TITLE
OCPBUGS-3612: configure-ovs: optionally generate configuration in /run

### PR DESCRIPTION
**Pull request by jcaamano**
Add support to configure via environment variable NM_CONN_SET_PATH where
are the configured NM keyfiles to reside. This path can be either the NM
etc or run paths and the environment variable can be configured through
a systemd dropin.

Change configure-ovs to generate the keyfiles in the NM run directory.

Setting up configure-ovs to work with the NM run path makes the
configuration ephemeral so that on a reboot there is no need to rollback any
configuration which has been problematic in some complex network
configurations.

Having this as a environment configuration variable gives the flexibility
to backport this solution and give users a choice in older releases.

### CODE REVIEW

**Comment by zeeke** [comment]
/lgtm

Job `ci/prow/e2e-alibabacloud` seems to be in bad shape [1] for reasons not related to this PR

[1] https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/pull-ci-openshift-machine-config-operator-master-e2e-alibabacloud

**Review by yuqi-zhang — APPROVED** [review]
Seems fine from a sanity check perspective, leaving an approval and a

/hold

for QE pre-merge testing

**Comment by jcaamano** [comment]
@yuqi-zhang it's been a while. Is this pre-merge testing somehting that should just happen or should we explicitly look for someone to do it?

/cc @zhaozhanqi

**Comment by yuqi-zhang** [comment]
I'm not sure the exact workflow. Is @zhaozhanqi the expected QE individual to verify this? If so we can reach out

**Comment by zhaozhanqi** [comment]
sorry for late reply due to holiday. using cluster-bot build one version `registry.build05.ci.openshift.org/ci-ln-sklj91k/release:latest`, and seems it worked well. 

```
# nmcli c s
NAME                UUID                                  TYPE           DEVICE 
ovs-if-br-ex        bd82f9bc-eebd-4c2d-b941-ae96eeb4df3e  ovs-interface  br-ex  
ovs-if-br-ex1       6c1be097-43aa-4164-aa16-619e452255b2  ovs-interface  br-ex1 
br-ex               a094fd1b-00a9-4119-aaff-cbcee4d79591  ovs-bridge     br-ex  
br-ex1              6d6e865f-dd7a-41e4-999a-4e0154e6ece0  ovs-bridge     br-ex1 
ovs-if-phys0        4f7890b6-da91-4008-a708-a2b50a7f60d3  ethernet       eno1   
ovs-if-phys1        cdd77835-1f5a-4993-a760-20addc0300e3  ethernet       eno2   
ovs-port-br-ex      cd90e31f-22f5-4094-96f4-0db3b61baa79  ovs-port       br-ex  
ovs-port-br-ex1     931dbe58-d16f-448a-a7d8-1d2dc26886de  ovs-port       br-ex1 
ovs-port-phys0      9e2902ee-6233-4d69-87cb-15806bba4f79  ovs-port       eno1   
ovs-port-phys1      29a49553-434e-43e7-be39-427f6e7d9d59  ovs-port       eno2   

```

**Comment by zeeke** [comment]
Hi @yuqi-zhang , can we remove the `hold` label here?

**Comment by yuqi-zhang** [comment]
Yes, sorry, in the future, the QE assignee can remove the hold after verification, since qe-approved label doesn't otherwise gate the PR from merging, which is why we added the hold in the first place.

/hold cancel

**Pull request merged**
